### ISSUE TITLE
TagHelper, RestrictChildren vs @if statement

### ISSUE THREAD

**Issue by lothistvan** [aspnet/Razor#697]
Version: rc1-final

My taghelper definition is

```
[HtmlTargetElement("panel")]
[RestrictChildren("panel-heading", "panel-body", "panel-footer")]
public class PanelTagHelper : TagHelper
{
	...
}
```

The problem is that I'd like to show the footer depends on something in the model:
```
<panel>
   <panel-body>
   ...
   </panel-body>
   @if(Model.NeedsFooter)
   {
   <panel-footer>
      ...
   </panel-footer>
   }
</panel>
```
The error message:

> The parent <panel> tag helper does not allow non-tag content. Only child tag helper(s) targeting tag name(s) 'panel-heading, panel-body, panel-footer' are allowed.

Please let us to use code block inside a taghelper (with restricted childrens).

Thanks, 
Istvan

**Comment by NTaylorMullen** [comment]
Sadly this is currently a restriction of the `RestrictChildren` attribute. However, I agree that we should definitely allow code when a parent has restricted children.

**Comment by lothistvan** [comment]
Thanks Taylor. Is there any workaround?

**Comment by NTaylorMullen** [comment]
You could bake the visibility functionality into the panel-footer tag helper or even build another `VisibilityTagHelper`.

That way it'd look something like:

```HTML
<panel>
  <panel-body>
  ...
  </panel-body>
  
  <panel-footer visible="Model.NeedsFooter">
  ...
  </panel-footer>
</panel>
```

**Comment by lothistvan** [comment]
Thanks again but this solution is not too nice for me. The panel(-xxx) taghelpers are renders a bootstrap panel used around the whole project (and company). I don't want to create dummy attributes as workaround. 
I commented out the restrictions and I hope I'll able to use code in the final version.

**Comment by Eilon** [comment]
@DamianEdwards / @NTaylorMullen how critical is this behavior? Backlog?

**Comment by DamianEdwards** [comment]
Yeah this seems overly restrictive. @NTaylorMullen any idea how much work to make this supported?

**Comment by NTaylorMullen** [comment]
@DamianEdwards @Eilon I took a quick look at the class responsible for restricting this behavior and it doesn't look like it'd require too much effort. Biggest time sink will be creating new tests for it.

Not too much effort equates to 2-3 days.

**Comment by DamianEdwards** [comment]
OK that's a little high for my comfort at this stage given the amount of other work we have remaining. Put it on the backlog.

**Comment by Eilon** [comment]
Backlog it is.
